### PR TITLE
(PLATFORM-2536) Escape or restrict vulnerable Oasis controllers

### PR DESCRIPTION
**Pull request by Grunny**
/cc @macbre @TK-999

### CODE REVIEW

File: extensions/wikia/CategorySelect/CategorySelectController.class.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
  * @author Kyle Florence <kflorence@wikia-inc.com>
  */
 
-class CategorySelectController extends WikiaController {
+class CategorySelectController extends WikiaService {
```
Comment (macbre) [inline extensions/wikia/CategorySelect/CategorySelectController.class.php]: Isn't `save` method called via AJAX request?

**Review by macbre — APPROVED** [review]

File: skins/oasis/modules/MenuButtonController.class.php [hunk 2]
```diff
@@ -168,7 +168,7 @@ public function executeIndex($data) {
 	/**
 	 * @param extraReturntoquery is a string which will be urlencoded and appended to the returntoquery. eg: "action=edit".
 	 */
-	public function createLoginURL($extraReturntoquery='') {
+	private function createLoginURL( $extraReturntoquery = '' ) {
 		global $wgTitle;
```
Comment (WikiaTech) [inline skins/oasis/modules/MenuButtonController.class.php:172]: MAJOR Remove this unused "$wgTitle" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

File: skins/oasis/modules/WikiaMediaCarouselController.class.php [hunk 3]
```diff
@@ -23,9 +23,9 @@ public function renderSlider() {
 		$this->enableScroll = ($dataCount <= $this->scrollLimit) ? false : true;
```
Comment (WikiaTech) [inline skins/oasis/modules/WikiaMediaCarouselController.class.php:23]: MINOR Remove the literal "true" boolean value. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1125)
MINOR Remove the literal "false" boolean value. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1125)

File: skins/oasis/modules/ErrorController.class.php [hunk 4]
```diff
@@ -1,10 +1,10 @@
 <?php
-class ErrorController extends WikiaController {
+class ErrorController extends WikiaService {
 
 	public function executeIndex($errors) {
 		if (isset($errors['controller'])) unset ($errors['controller']);
```
Comment (WikiaTech) [inline skins/oasis/modules/ErrorController.class.php:5]: MAJOR Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
MINOR 2 statements were found on this line. Reformat the code to have only one statement per line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS122)

File: skins/oasis/modules/ErrorController.class.php [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 		if (isset($errors['method'])) unset ($errors['method']);
```
Comment (WikiaTech) [inline skins/oasis/modules/ErrorController.class.php:6]: MAJOR Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
MINOR 2 statements were found on this line. Reformat the code to have only one statement per line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS122)

File: extensions/wikia/CategorySelect/CategorySelectController.class.php [hunk 6]
```diff
@@ -8,7 +8,7 @@
...
 	const CACHE_TTL_AJAX = 360; // 1 hour
```
Comment (WikiaTech) [inline extensions/wikia/CategorySelect/CategorySelectController.class.php:12]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

File: extensions/wikia/CategorySelect/CategorySelectController.class.php [hunk 7]
```diff
@@ -8,7 +8,7 @@
...
 	const CACHE_TTL_MEMCACHE = 86400; // 1 day
```
Comment (WikiaTech) [inline extensions/wikia/CategorySelect/CategorySelectController.class.php:13]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 27 issues
- MAJOR 8 major
- MINOR 16 minor
- INFO 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [MenuButtonController.class.php#L19](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L19): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
2. MAJOR [MenuButtonController.class.php#L40](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L40): The Cyclomatic Complexity of this function "executeIndex" is 28 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
3. MAJOR [MenuButtonController.class.php#L136](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L136): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
4. MAJOR [MenuButtonController.class.php#L177](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L177): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
5. MAJOR [MenuButtonController.class.php#L195](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L195): This case's code block is the same as the block for the case on line 192. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1871)
6. MINOR [CategorySelectController.class.php#L52](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/extensions/wikia/CategorySelect/CategorySelectController.class.php#L52): Define a constant instead of duplicating this literal "categories" 15 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
7. MINOR [CategorySelectController.class.php#L199](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/extensions/wikia/CategorySelect/CategorySelectController.class.php#L199): Define a constant instead of duplicating this literal "error" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
8. MINOR [ArticleInterlangController.class.php#L38](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/ArticleInterlangController.class.php#L38): Define a constant instead of duplicating this literal "class" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
9. MINOR [MenuButtonController.class.php#L45](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L45): Define a constant instead of duplicating this literal "action" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
10. MINOR [MenuButtonController.class.php#L116](https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/skins/oasis/modules/MenuButtonController.class.php#L116): Define a constant instead of duplicating this literal "class" 6 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CategorySelect/CategorySelectController.class.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/wikia/CategorySelect/CategorySelectController.class.php]: `save` method is indeed called via ajax - here: https://github.com/Wikia/app/blob/85205157ef0f5f697438128d9b0b9aa7982989f4/extensions/wikia/CategorySelect/js/CategorySelect.view.js#L83

We should only limit accessibility of the vulnerable method.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

**Review by mszabo-wikia — APPROVED** [review]
[LGTM](https://lgtm.in/i/iTRALLFbq)

**Pull request merged**